src/logic/treeherder.ts

```diff
@@ -92,6 +109,78 @@ export async function fetchCompareOverTimeResults({
   return response.json() as Promise<CompareResultsItem[]>;
 }
 
+// This API returns the subtests results of a particular comparison result between 2 revisions.
+export async function fetchSubtestsCompareResults({
+  baseRev,
+  baseRepo,
+  newRev,
+  newRepo,
+  framework,
+  parentSignature,
+}: FetchSubtestsProps) {
+  const searchParams = new URLSearchParams({
+    base_repository: baseRepo,
+    base_revision: baseRev,
+    new_repository: newRepo,
+    new_revision: newRev,
+    framework: String(framework),
+    parent_signature: parentSignature,
+  });
+
+  const response = await fetch(
+    `${treeherderBaseURL}/api/perfcompare/results/?${searchParams.toString()}`,
+  );
+  if (!response.ok) {
+    if (response.status === 400) {
+      throw new Error(
+        `Error when requesting treeherder: ${await response.text()}`,
+      );
+    } else {
+      throw new Error(
+        `Error when requesting treeherder: (${response.status}) ${response.statusText}`,
+      );
+    }
+  }
+
+  return response.json() as Promise<CompareResultsItem[]>;
+}
+
+// This API returns the subtests results of a particular comparison result between 2 revisions.
+export async function fetchSubtestsCompareOverTimeResults({
+  baseRepo,
+  newRev,
+  newRepo,
+  framework,
+  interval,
+  parentSignature,
+}: FetchSubtestsOverTimeProps) {
+  const searchParams = new URLSearchParams({
+    base_repository: baseRepo,
+    new_repository: newRepo,
+    new_revision: newRev,
+    framework: String(framework),
+    interval: String(interval),
+    parent_signature: parentSignature,
+  });
+
+  const response = await fetch(
+    `${treeherderBaseURL}/api/perfcompare/results/?${searchParams.toString()}`,
+  );
+  if (!response.ok) {
+    if (response.status === 400) {
+      throw new Error(
+        `Error when requesting treeherder: ${await response.text()}`,
+      );
+    } else {
+      throw new Error(
+        `Error when requesting treeherder: (${response.status}) ${response.statusText}`,
+      );
+    }
+  }
+
+  return response.json() as Promise<CompareResultsItem[]>;
+}
+
```

Comment:
All the functions here are very similar, only the searchParams is different in all these functions.
We could probably extract a (non exported) function taking the `searchParams` as a parameter, doing the fetch, so that all the functions would be smaller.
Optional for this PR, this could be a follow-up.

Reply:
I made a separate [PR](https://github.com/mozilla/perfcompare/pull/663) for this.